client/finality-grandpa/rpc/src/lib.rs

```diff
@@ -133,22 +228,57 @@ mod tests {
 		}
 	}
 
+	lazy_static::lazy_static! {
+		static ref EXECUTOR: executor::ThreadPool = executor::ThreadPool::new()
+			.expect("Failed to create thread pool executor for tests");
+	}
+
+	// These are lifted from the jsonrpc_pubsub crate tests
+	pub struct TestTaskExecutor;
```

Comment:
There is already an `Executor` for tests in `crate::testing::TaskExecutor`.

Reply:
This is private in a different crate through (sc-rpc)

Reply:
IMHO Totally fine to make it either public or feature-gated (under `test-helpers`) - much better than to have two copies of the same thing within the same repo.